frontend/src/lib/lemon-ui/LemonTree/LemonTree.tsx

```diff
@@ -558,31 +558,36 @@ const LemonTreeNode = forwardRef<HTMLDivElement, LemonTreeNodeProps>(
                                             button
                                         )}
 
-                                        {itemSideAction && !isEmptyFolder && size === 'default' && (
-                                            <DropdownMenu>
-                                                <DropdownMenuTrigger asChild>
-                                                    <ButtonPrimitive
-                                                        iconOnly
-                                                        isSideActionRight
-                                                        className="z-2 shrink-0 motion-safe:transition-opacity duration-[50ms] group-hover/lemon-tree-button-group:opacity-100 aria-expanded:opacity-100 h-[var(--lemon-tree-button-height)]"
-                                                    >
-                                                        {itemSideActionIcon?.(item) ?? (
-                                                            <IconEllipsis className="size-3 text-tertiary" />
-                                                        )}
-                                                    </ButtonPrimitive>
-                                                </DropdownMenuTrigger>
-
-                                                {/* The Dropdown content menu */}
-                                                <DropdownMenuContent
-                                                    loop
-                                                    align="end"
-                                                    side="bottom"
-                                                    className="max-w-[250px]"
-                                                >
-                                                    {itemSideAction(item)}
-                                                </DropdownMenuContent>
-                                            </DropdownMenu>
-                                        )}
+                                        {itemSideAction &&
+                                            itemSideAction(item) !== undefined &&
```

Comment:
Not sure about the best pattern here without blowing up any defaults but I needed something where I could override sideaction through icon